### PR TITLE
Fix: Train loop config validation was run during `trainer.predict`

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Fixes:
While predicting, it was verifying the train_loop configuration.

**Only necessary for the bug-fix release branch**

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by carmocca — APPROVED** [review]
LGTM

Do you want to also open a PR to master with these changes and resolve this `TODO`?

https://github.com/PyTorchLightning/pytorch-lightning/blob/6a1414681107cf77d1ebeba86359df7b6668bf49/pytorch_lightning/trainer/configuration_validator.py#L43

**Comment by kaushikb11** [comment]
@carmocca Sure.

Wondering where should I do the Changelog update here.

File: tests/trainer/test_trainer.py [hunk 1]
```diff
@@ -1850,3 +1850,35 @@ def compare_optimizers():
...
+
+    model = TestModel()
+    datamodule = TestLightningDataModule(dataloaders)
+
+    trainer = Trainer(default_root_dir=tmpdir)
+
+    if datamodule:
+        results = trainer.predict(model, datamodule=datamodule)
+    else:
+        results = trainer.predict(model, dataloaders=dataloaders)
```
Comment (Borda) [inline tests/trainer/test_trainer.py:1876]: probably missing some parametrization for case with and without datamodule

**Review by Borda — CHANGES_REQUESTED** [review]

**Comment by Borda** [comment]
> Wondering where should I do the Changelog update here.

in Added section :]

File: tests/trainer/test_trainer.py (hunk 1 above)
Comment (kaushikb11) [reply tests/trainer/test_trainer.py:1876]: @Borda Do you mean like this?
```python
@pytest.mark.parametrize('datamodule', [False, True])
```

Comment (kaushikb11) [reply tests/trainer/test_trainer.py]: @Borda Updated!

Comment (kaushikb11) [reply tests/trainer/test_trainer.py]: Good catch

**Review by justusschock — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Comment by kaushikb11** [comment]
@Borda Your commit won't work. `dataloaders` is not defined. Reverting back.

**Comment by Borda** [comment]
> @Borda Your commit won't work. `dataloaders` is not defined. Reverting back.

it shall be defined in the parameter, right?

**Comment by Borda** [comment]
what is the difference to #6543?
